### PR TITLE
Random Vogon fixes

### PR DESCRIPTION
**Pull request by vanam**
Hi,
firstly, I would like to thank you for this amazing demo.

There are several minor improvements which might make your demo one bit more amazing.

So Long, and Thanks for All the Fish

### CODE REVIEW

**Comment by vanam** [comment]
Strange thing happens when I run `./gradlew resolveAndLockAll --write-locks`. Every time it messes up some `org.jetbrains.kotlin` dependencies. 

It turns
```
org.jetbrains.kotlin:kotlin-stdlib-common:2.0.0=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.9.24=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.24=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
```
to
```
org.jetbrains.kotlin:kotlin-stdlib-common:2.0.0=compileClasspath,testCompileClasspath
org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.9.24=compileClasspath,testCompileClasspath
org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.24=compileClasspath,testCompileClasspath
```
and then `./gradlew build` fails:
```
Execution failed for task ':tea-service:cyclonedxBom'.
> Could not resolve all dependencies for configuration ':tea-service:runtimeClasspath'.
   > Resolved 'org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.24' which is not part of the dependency lock state
   > Resolved 'org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.9.24' which is not part of the dependency lock state
   > Resolved 'org.jetbrains.kotlin:kotlin-stdlib-common:2.0.0' which is not part of the dependency lock state

```

I have to manualy remove this change from changelist. Is it also happening to you?

File: .github/dependabot.yml
Comment (jonatan-ivanov) [inline .github/dependabot.yml:1]: I'm not sure what is the goal with this file. As far as I know Dependabot cannot deal with gradle lock files and this file also contain a section for Maven that this project does not have, please revert.

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -21,6 +21,6 @@ jobs:
         cache: 'gradle'
         check-latest: true
     - name: Validate Gradle wrapper
-      uses: gradle/wrapper-validation-action@v2
+      uses: gradle/actions/wrapper-validation@v3
```
Comment (jonatan-ivanov) [inline .github/workflows/gradle.yml:24]: Good catch, do you want to do this in a separate PR?

File: build.gradle [hunk 2]
```diff
@@ -3,7 +3,7 @@ ext.buildTime = new Date().toInstant()
 buildscript {
     ext.springBootVersion = '3.+'
     ext.springCloudVersion = '2023.+'
-    ext.springDataVersion = '2023.+'
+    ext.springDataVersion = '2024.+'
```
Comment (jonatan-ivanov) [inline build.gradle:6]: Since this change can be trickier than it seems, I would like to do this separately.

File: core/gradle.lockfile
Comment (jonatan-ivanov) [inline core/gradle.lockfile:1]: Please revert lock file changes, I would like to deal with upgrading existing dependencies with automatic version bumps and manual version bumps separately.

File: docker-compose.yml [hunk 3]
```diff
@@ -82,22 +82,22 @@ services:
...
+            - "3100:3100"
     maildev:
         container_name: maildev
         image: maildev/maildev:2.1.0 # https://hub.docker.com/r/maildev/maildev/tags
         extra_hosts: [ 'host.docker.internal:host-gateway' ]
         ports:
-            - 3001:1080
-            - 25:1025
+            - "3001:1080"
+            - "25:1025"
```
Comment (jonatan-ivanov) [inline docker-compose.yml:100]: If my memory serves me well, not using `"` is intentional here since Docker Compose has/had issues with it for lower port numbers.
Unless this was fixed and you have a reason why quotes should be here, I would prefer not to include them.

File: eureka/src/main/resources/application.properties [hunk 4]
```diff
@@ -2,6 +2,8 @@ spring.application.name=eureka
 spring.application.org=teahouse
 spring.profiles.default=local
 server.port=8761
+eureka.client.register-with-eureka=false
```
Comment (jonatan-ivanov) [inline eureka/src/main/resources/application.properties:5]: Registering Eureka to itself is intentional, please revert this change.

File: eureka/src/main/resources/logback-access.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
     <jmxConfigurator />
     <statusListener class="ch.qos.logback.core.status.OnConsoleStatusListener" />
     <property scope="local" name="accessLogPattern" value="%date{yyyy-MM-dd HH:mm:ss.SSS} traceid=&quot;%i{x-b3-traceid}&quot; spanid=&quot;%i{x-b3-spanid}&quot; parentspanid=&quot;%i{x-b3-parentspanid}&quot; method=&quot;%m&quot; url=&quot;%U&quot; query=&quot;%q&quot; protocol=&quot;%H&quot; statusCode=&quot;%s&quot; duration=&quot;%D&quot; bytesSent=&quot;%b&quot; remoteIp=&quot;%a&quot; remoteHost=&quot;%h&quot; localIp=&quot;%A&quot; localServer=&quot;%v&quot; localPort=&quot;%localPort&quot; userAgent=&quot;%i{User-Agent}&quot;" />
-    <property scope="local" name="fileName" value="${spring.application.home:-.}/logs/water-service.access.log" />
+    <property scope="local" name="fileName" value="${spring.application.home:-.}/logs/eureka.access.log" />
```
Comment (jonatan-ivanov) [inline eureka/src/main/resources/logback-access.xml:6]: Good catch, do you want to do this (with the other `logback-access` filename fixes) in a separate PR?

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-bin.zip
```
Comment (jonatan-ivanov) [inline gradle/wrapper/gradle-wrapper.properties:3]: Do you want to do this in a separate PR?

File: spring-boot-admin/src/main/java/org/example/teahouse/sba/SbaApplication.java [hunk 7]
```diff
@@ -6,12 +6,10 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.context.metrics.buffering.BufferingApplicationStartup;
-import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
 import org.springframework.context.annotation.Import;
 import org.springframework.context.annotation.PropertySource;
 
 @EnableAdminServer
-@EnableDiscoveryClient
```
Comment (jonatan-ivanov) [inline spring-boot-admin/src/main/java/org/example/teahouse/sba/SbaApplication.java:14]: Good catch, do you want to do this (+ fixing the other files with this annotation) in a separate PR?
If so please mention in the commit message that according to the Spring Cloud docs:
>@EnableDiscoveryClient is no longer required. You can put a DiscoveryClient implementation on the classpath to cause the Spring Boot application to register with the service discovery server.

https://docs.spring.io/spring-cloud-commons/reference/4.1/index.html

File: tealeaf-service/tealeaf-api/build.gradle [hunk 8]
```diff
@@ -1,3 +1,3 @@
 dependencies {
-    implementation 'jakarta.validation:jakarta.validation-api:3.0.2'
+    implementation 'jakarta.validation:jakarta.validation-api:3.1.0'
```
Comment (jonatan-ivanov) [inline tealeaf-service/tealeaf-api/build.gradle:2]: Let me deal with this separately after upgrading the dependencies without this change.

**Review by jonatan-ivanov — COMMENTED** [review]
Thank you for the PR, I'm glad you like the project.
This PR does a lot of things, some I don't want to merge (right now/as-is) and some I would like to merge in but not all in one PR but separately.

If you don't want to open separate PRs (it's work), please let me know and I will cherry pick your commits and merge those separately.

**Comment by jonatan-ivanov** [comment]
>Strange thing happens when I run ./gradlew resolveAndLockAll --write-locks. Every time it messes up some org.jetbrains.kotlin dependencies.
> [...]
>I have to manualy remove this change from changelist. Is it also happening to you?

Unfortunately it does, I think this is caused by okhttp that uses the Kotlin SDK, it's on my list to fix it, I created an issue to track it: https://github.com/jonatan-ivanov/teahouse/issues/13
I'm not a fan of having the Kotlin SDK on the classpath anyways so this can mean replacing okhttp (maybe OpenFeign too) for the new Interface Clients in Spring Framework but I need to check if instrumentation works with the customizations I have and if there is Logbook support available.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (vanam) [reply .github/workflows/gradle.yml:24]: See #15

File: docker-compose.yml (hunk 3 above)
Comment (vanam) [reply docker-compose.yml:100]: Created #16



> HOST:CONTAINER should always be specified as a (quoted) string, to avoid conflicts with yaml base-60 float

https://docs.docker.com/compose/compose-file/05-services/#ports

File: eureka/src/main/resources/logback-access.xml (hunk 5 above)
Comment (vanam) [reply eureka/src/main/resources/logback-access.xml:6]: Created #17

File: gradle/wrapper/gradle-wrapper.properties (hunk 6 above)
Comment (vanam) [reply gradle/wrapper/gradle-wrapper.properties:3]: Created #14

File: spring-boot-admin/src/main/java/org/example/teahouse/sba/SbaApplication.java (hunk 7 above)
Comment (vanam) [reply spring-boot-admin/src/main/java/org/example/teahouse/sba/SbaApplication.java:14]: See #18

File: .github/dependabot.yml
Comment (vanam) [reply .github/dependabot.yml:1]: It is not going to create a valid PR, but might be useful as a reminder that there is a newer version. (That is how I discovered that there is a newer Spring Data and Spring Validation).

Yeah, the maven section is useless and docker section is missing.

**Comment by vanam** [comment]
I guess we can close this PR now.

File: .github/dependabot.yml
Comment (jonatan-ivanov) [reply .github/dependabot.yml:1]: I don't necessarily upgrade versions immediately after they were released (except if it contains a fix/feature I need). I think usually I wait for the Spring Cloud release train to happen before upgrading Boot and Data. Btw you might find this helpful: https://calendar.spring.io/

Even though I'm usually aware of these new versions, I think I can "automate" this without the help of Depandabot by just using `latest.release`. I can make this change and let's see how it works (I can always pin versions as needed).

**Comment by jonatan-ivanov** [comment]
Thank you very much!

**Comment by jonatan-ivanov** [comment]
Superseded by #14, #15 ,#16, #17, #18.

**Comment by jonatan-ivanov** [comment]
I think the merged PRs and these commits:
- https://github.com/jonatan-ivanov/teahouse/commit/e4a6cfa4f50b84ae71f96912b8a6c124af4ecec1
- https://github.com/jonatan-ivanov/teahouse/commit/a84d86a2da1c90ba5b9c382b0b4a975425737f65
should cover the changes in this PR, please let me know if this is not the case.

Thank you very much once again for the contributions!
